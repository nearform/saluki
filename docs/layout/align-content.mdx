import { AlignContent, AlignContentBox } from './'

# Align Content

---

## Example

<AlignContent>
  <AlignContentBox light />
  <AlignContentBox />
</AlignContent>

```jsx
const Container = styled.div`
  ${size.height.large}
  ${layout.wrap.wrap}
  ${layout.display.flex}
  ${layout.alignContent.between}
  ${background.yellow}
`

const Box = styled.div`
  ${size.height.small}
  ${size.width.full}
  ${({ light }) => (light ? background.red : background.black)};
`
```

## Rules

```js
layout.alignContent.start
layout.alignContent.end
layout.alignContent.center
layout.alignContent.between
layout.alignContent.around
```

## Values

```js
{
  layout: {
    alignContent: {
      start: 'flex-start',
      end: 'flex-end',
      center: 'center',
      between: 'space-between',
      around: 'space-around'
    }
  }
}
```
