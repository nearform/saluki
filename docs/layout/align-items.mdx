import { AlignItems, Box } from './'

# Align Items

---

## Example

<AlignItems>
  <Box light />
  <Box />
  <Box light />
</AlignItems>

```jsx
const Container = styled.div`
  ${size.height.large}
  ${layout.display.flex}
  ${layout.alignItems.center}
  ${background.color.yellow}
`

const Box = styled.div`
  ${size.height.medium}
  ${size.width.third}
  ${({ light }) => (light ? background.color.red : background.color.black)}
`
```

## Rules

```js
layout.alignItems.start
layout.alignItems.end
layout.alignItems.stretch
layout.alignItems.center
layout.alignItems.baseline
```
