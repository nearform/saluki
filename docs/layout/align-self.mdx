import { AlignSelf, AlignSelfBox } from './'

# Align Self

---

## Example

<AlignSelf>
  <AlignSelfBox stretch />
  <AlignSelfBox />
  <AlignSelfBox stretch />
  <AlignSelfBox />
</AlignSelf>

```jsx
const Container = styled.div`
  ${size.height.large}
  ${layout.display.flex}
  ${background.color.yellow}
`

const Box = styled.div`
  ${({ stretch }) => (stretch ? layout.alignSelf.stretch : size.height.medium)}
  ${size.width.quarter}
  ${({ stretch }) => (stretch ? background.color.red : background.color.black)}
`
```

## Rules

```js
layout.alignSelf.start
layout.alignSelf.end
layout.alignSelf.stretch
layout.alignSelf.center
layout.alignSelf.auto
```
