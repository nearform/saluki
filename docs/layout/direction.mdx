import { Direction, DirectionBox } from './'

# Flex Direction

---

## Example

<Direction>
  <DirectionBox light />
  <DirectionBox />
  <DirectionBox light />
</Direction>

```jsx
const Container = styled.div`
  ${layout.display.flex}
  ${layout.direction.rowReverse}
`

const Box = styled.div`
  ${size.height.medium}
  ${size.width.sixth}
  ${({ light }) => (light ? background.red : background.black)};
`
```

## Rules

```js
layout.direction.row
layout.direction.rowReverse
layout.direction.column
layout.direction.columnReverse
```

## Defaults

```js
{
  layout: {
    direction: {
      row: 'row',
      rowReverse: 'row-reverse',
      column: 'column',
      columnReverse: 'column-reverse'
    }
  }
}
```
