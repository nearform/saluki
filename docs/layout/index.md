# Layout

---

## Rules

```js
layout.display.block
layout.display.inlineBlock
layout.display.flex
layout.display.inlineFlex

layout.float.left
layout.float.right
layout.float.none

layout.position.fixed
layout.position.static
layout.position.relative
layout.position.absolute
layout.position.sticky

layout.overflow.hidden
layout.overflow.visible
layout.overflow.scroll
layout.overflow.auto

layout.direction.row
layout.direction.rowReverse
layout.direction.column
layout.direction.columnReverse

layout.alignItems.start
layout.alignItems.end
layout.alignItems.stretch
layout.alignItems.center
layout.alignItems.baseline

layout.wrap.noWrap
layout.wrap.wrap
layout.wrap.reverse

layout.alignContent.start
layout.alignContent.end
layout.alignContent.center
layout.alignContent.between
layout.alignContent.around

layout.alignSelf.start
layout.alignSelf.end
layout.alignSelf.stretch
layout.alignSelf.center
layout.alignSelf.auto

layout.justifyContent.start
layout.justifyContent.end
layout.justifyContent.center
layout.justifyContent.between
layout.justifyContent.around
```

## Defaults

```js
{
  display: {
    flex: 'flex',
    inlineFlex: 'inline-flex',
    block: 'block',
    inlineBlock: 'inline-block'
  },
  float: {
    left: 'left',
    right: 'right',
    none: 'none'
  },
  position: {
    fixed: 'fixed',
    static: 'static',
    relative: 'relative',
    absolute: 'absolute',
    sticky: 'sticky'
  },
  overflow: {
    hidden: 'hidden',
    visible: 'visible',
    scroll: 'scroll',
    auto: 'auto'
  },
  direction: {
    row: 'row',
    rowReverse: 'row-reverse',
    column: 'column',
    columnReverse: 'column-reverse'
  },
  alignItems: {
    start: 'flex-start',
    end: 'flex-end',
    stretch: 'stretch',
    center: 'center',
    baseline: 'baseline'
  },
  wrap: {
    noWrap: 'nowrap',
    wrap: 'wrap',
    reverse: 'wrap-reverse'
  },
  alignContent: {
    start: 'flex-start',
    end: 'flex-end',
    center: 'center',
    between: 'space-between',
    around: 'space-around'
  },
  alignSelf: {
    start: 'flex-start',
    end: 'flex-end',
    center: 'center',
    auto: 'auto',
    stretch: 'stretch'
  },
  justifyContent: {
    start: 'flex-start',
    end: 'flex-end',
    center: 'center',
    between: 'space-between',
    around: 'space-around'
  }
}
```
