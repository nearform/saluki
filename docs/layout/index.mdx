# Layout

---

## Rules

```js
layout.display.block
layout.display.inlineBlock
layout.display.flex
layout.display.inlineFlex

layout.flex.grow
layout.flex.noGrow
layout.flex.shrink
layout.flex.noShrink
layout.flex.none
layout.flex.one
layout.flex.auto
layout.flex.initial

layout.float.left
layout.float.right
layout.float.none

layout.position.fixed
layout.position.static
layout.position.relative
layout.position.absolute
layout.position.sticky

layout.overflow.hidden
layout.overflow.visible
layout.overflow.scroll
layout.overflow.auto

layout.direction.row
layout.direction.rowReverse
layout.direction.column
layout.direction.columnReverse

layout.alignItems.start
layout.alignItems.end
layout.alignItems.stretch
layout.alignItems.center
layout.alignItems.baseline

layout.wrap.noWrap
layout.wrap.wrap
layout.wrap.reverse

layout.alignContent.start
layout.alignContent.end
layout.alignContent.center
layout.alignContent.between
layout.alignContent.around

layout.alignSelf.start
layout.alignSelf.end
layout.alignSelf.stretch
layout.alignSelf.center
layout.alignSelf.auto

layout.justifyContent.start
layout.justifyContent.end
layout.justifyContent.center
layout.justifyContent.between
layout.justifyContent.around
```
