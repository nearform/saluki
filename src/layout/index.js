export const defaultRules = {
  display: {
    flex: 'flex',
    inlineFlex: 'inline-flex',
    block: 'block',
    inlineBlock: 'inline-block'
  },
  float: {
    left: 'left',
    right: 'right',
    none: 'none'
  },
  position: {
    fixed: 'fixed',
    static: 'static',
    relative: 'relative',
    absolute: 'absolute',
    sticky: 'sticky'
  },
  overflow: {
    hidden: 'hidden',
    visible: 'visible',
    scroll: 'scroll',
    auto: 'auto'
  },
  direction: {
    row: 'row',
    rowReverse: 'row-reverse',
    column: 'column',
    columnReverse: 'column-reverse'
  },
  alignItems: {
    start: 'flex-start',
    end: 'flex-end',
    stretch: 'stretch',
    center: 'center',
    baseline: 'baseline'
  },
  wrap: {
    noWrap: 'nowrap',
    wrap: 'wrap',
    reverse: 'wrap-reverse'
  },
  alignContent: {
    start: 'flex-start',
    end: 'flex-end',
    center: 'center',
    between: 'space-between',
    around: 'space-around'
  },
  alignSelf: {
    start: 'flex-start',
    end: 'flex-end',
    center: 'center',
    auto: 'auto',
    stretch: 'stretch'
  },
  justifyContent: {
    start: 'flex-start',
    end: 'flex-end',
    center: 'center',
    between: 'space-between',
    around: 'space-around'
  }
}

function generateRules(rules, cssProp) {
  const newRules = {}

  for (const [key, value] of Object.entries(rules)) {
    newRules[key] = {
      [cssProp]: value
    }
  }

  return newRules
}

export default function(rules) {
  return {
    display: generateRules(rules.display, 'display'),
    float: generateRules(rules.float, 'float'),
    position: generateRules(rules.position, 'position'),
    overflow: generateRules(rules.overflow, 'overflow'),
    direction: generateRules(rules.direction, 'flex-direction'),
    alignItems: generateRules(rules.alignItems, 'align-items'),
    wrap: generateRules(rules.wrap, 'flex-wrap'),
    alignContent: generateRules(rules.alignContent, 'align-content'),
    alignSelf: generateRules(rules.alignSelf, 'align-self'),
    justifyContent: generateRules(rules.justifyContent, 'justify-content')
  }
}
