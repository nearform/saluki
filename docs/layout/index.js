import styled from 'styled-components/macro'
import { layout, background, size, spacing } from '../../src'

export const Container = styled.div`
  ${layout.display.flex}
`

export const Direction = styled.div`
  ${layout.display.flex}
  ${layout.direction.rowReverse}
`

export const Box = styled.div`
  ${size.height.medium}
  ${size.width.half}
  ${({ light }) => (light ? background.color.red : background.color.black)}
`

export const FlexBox = styled.div`
  ${size.height.medium}
  ${layout.flex.one}
  ${({ bg }) => background.color[bg]}
`

export const DirectionBox = styled.div`
  ${size.height.medium}
  ${size.width.third}
  ${({ light }) => (light ? background.color.red : background.color.black)}
`
export const AlignItems = styled.div`
  ${size.height.large}
  ${layout.display.flex}
  ${layout.alignItems.center}
  ${background.color.yellow}
`

export const AlignContent = styled.div`
  ${size.height.large}
  ${layout.wrap.wrap}
  ${layout.display.flex}
  ${layout.alignContent.between}
  ${background.color.yellow}
`

export const AlignContentBox = styled.div`
  ${size.height.small}
  ${size.width.full}
  ${({ light }) => (light ? background.color.red : background.color.black)}
`

export const Wrap = styled.div`
  ${layout.wrap.wrap}
  ${layout.display.flex}
`

export const AlignSelf = styled.div`
  ${size.height.large}
  ${layout.display.flex}
  ${background.color.yellow}
`

export const AlignSelfBox = styled.div`
  ${({ stretch }) => (stretch ? layout.alignSelf.stretch : size.height.medium)}
  ${size.width.quarter}
  ${({ stretch }) => (stretch ? background.color.red : background.color.black)}
`

export const JustifyContent = styled.div`
  ${layout.display.flex}
  ${layout.justifyContent.around}
`

export const JustifyContentBox = styled.div`
  ${size.height.medium}
  ${size.width.third}
  ${({ light }) => (light ? background.color.red : background.color.black)}
`

export const FloatContainer = styled.div`
  ${layout.display.block}
  ${size.height.large}
`

export const Float = styled.div`
  ${size.width.half}
  ${size.height.medium}
  ${layout.float.right}
  ${background.color.red}
`

export const OverflowContainer = styled.div`
  ${layout.display.block}
  ${size.height.large}
`

export const Overflow = styled.div`
  ${layout.overflow.scroll}
  ${size.height.medium}
`

export const PositionContainer = styled.div`
  ${spacing.padding.vertical.medium}
`

export const Position = styled.div`
  ${layout.position.fixed}
`
