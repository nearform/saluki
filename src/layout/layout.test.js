import layout, { defaultRules } from './'

describe('layout', () => {
  it('should return the default config', () => {
    expect(layout(defaultRules)).toMatchSnapshot()
  })
})
