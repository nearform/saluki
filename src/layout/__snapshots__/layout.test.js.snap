// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should return the default config 1`] = `
Object {
  "alignContent": Object {
    "around": Object {
      "align-content": "space-around",
    },
    "between": Object {
      "align-content": "space-between",
    },
    "center": Object {
      "align-content": "center",
    },
    "end": Object {
      "align-content": "flex-end",
    },
    "start": Object {
      "align-content": "flex-start",
    },
  },
  "alignItems": Object {
    "baseline": Object {
      "align-items": "baseline",
    },
    "center": Object {
      "align-items": "center",
    },
    "end": Object {
      "align-items": "flex-end",
    },
    "start": Object {
      "align-items": "flex-start",
    },
    "stretch": Object {
      "align-items": "stretch",
    },
  },
  "alignSelf": Object {
    "auto": Object {
      "align-self": "auto",
    },
    "center": Object {
      "align-self": "center",
    },
    "end": Object {
      "align-self": "flex-end",
    },
    "start": Object {
      "align-self": "flex-start",
    },
    "stretch": Object {
      "align-self": "stretch",
    },
  },
  "direction": Object {
    "column": Object {
      "flex-direction": "column",
    },
    "columnReverse": Object {
      "flex-direction": "column-reverse",
    },
    "row": Object {
      "flex-direction": "row",
    },
    "rowReverse": Object {
      "flex-direction": "row-reverse",
    },
  },
  "display": Object {
    "block": Object {
      "display": "block",
    },
    "flex": Object {
      "display": "flex",
    },
    "inlineBlock": Object {
      "display": "inline-block",
    },
    "inlineFlex": Object {
      "display": "inline-flex",
    },
  },
  "float": Object {
    "left": Object {
      "float": "left",
    },
    "none": Object {
      "float": "none",
    },
    "right": Object {
      "float": "right",
    },
  },
  "justifyContent": Object {
    "around": Object {
      "justify-content": "space-around",
    },
    "between": Object {
      "justify-content": "space-between",
    },
    "center": Object {
      "justify-content": "center",
    },
    "end": Object {
      "justify-content": "flex-end",
    },
    "start": Object {
      "justify-content": "flex-start",
    },
  },
  "overflow": Object {
    "auto": Object {
      "overflow": "auto",
    },
    "hidden": Object {
      "overflow": "hidden",
    },
    "scroll": Object {
      "overflow": "scroll",
    },
    "visible": Object {
      "overflow": "visible",
    },
  },
  "position": Object {
    "absolute": Object {
      "position": "absolute",
    },
    "fixed": Object {
      "position": "fixed",
    },
    "relative": Object {
      "position": "relative",
    },
    "static": Object {
      "position": "static",
    },
    "sticky": Object {
      "position": "sticky",
    },
  },
  "wrap": Object {
    "noWrap": Object {
      "flex-wrap": "nowrap",
    },
    "reverse": Object {
      "flex-wrap": "wrap-reverse",
    },
    "wrap": Object {
      "flex-wrap": "wrap",
    },
  },
}
`;
